### PR TITLE
Feature/attendance

### PR DESCRIPTION
**Pull request by Naomilounsbury**
got the attend button working
note: I added a return in api user-routes login because the session storage wasnt working properly.

### CODE REVIEW

File: controllers/api/activity-routes.js [hunk 1]
```diff
@@ -146,13 +146,16 @@ router.post("/", (req, res) => {
...
 // POST route for attending an activity
+
 // POST /api/activity/attend
 router.post("/attend", (req, res) => {
   Attendance.create({
     user_id: req.session.user_id,
     activity_id: req.body.activity_id,
   })
-    .then((dbActivityData) => dbActivityData)
+    .then(() => {
```
Comment (jdpasternak) [inline controllers/api/activity-routes.js:156]: We don't render HTML from an API route. What is the purpose of this?

File: controllers/api/user-routes.js [hunk 2]
```diff
@@ -167,7 +167,7 @@ router.post("/login", (req, res) => {
         req.session.username = dbUserData.username;
         req.session.loggedIn = true;
 
-        res.json({ user: dbUserData, message: "You are now logged in!" });
+        return res.json({ user: dbUserData, message: "You are now logged in!" });
```
Comment (jdpasternak) [inline controllers/api/user-routes.js:170]: Should not return anything here. `res.json()` is used to provide the data back to the client and doesn't return anything useful

File: controllers/homepage-routes.js [hunk 3]
```diff
@@ -199,11 +199,12 @@ router.get("/activity/:id", withAuth, (req, res) => {
     ],
   })
     .then((dbActivityData) => {
-      console.log(dbActivityData.get({ plain: true }));
+      console.log(dbActivityData.get({ plain: true }), "HEY HO");
```
Comment (jdpasternak) [inline controllers/homepage-routes.js:202]: What is the purpose of this?

File: controllers/homepage-routes.js [hunk 4]
```diff
@@ -199,11 +199,12 @@ router.get("/activity/:id", withAuth, (req, res) => {
...
       res.render("activity", {
         activity: dbActivityData.get({ plain: true }),
         user_id: req.session.user_id,
         loggedIn: req.session.loggedIn,
+        count: dbActivityData.attending.length,
```
Comment (jdpasternak) [inline controllers/homepage-routes.js:207]: This is called something different elsewhere. We should stick to same naming of all model attributes.  Refer to `activity-routes.js` lines 95-98 for an example of how to get this data and how to return it as part of the rest of the body

File: public/javascript/attend.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+const confirmAttendButtonHandler = async (event) => {
+  event.preventDefault();
+
+  const activityId =
+    window.location.pathname.split("/")[
+      window.location.pathname.split("/").length - 1
+    ];
+
+  const userId = window.sessionStorage.user_id;
```
Comment (jdpasternak) [inline public/javascript/attend.js:14]: You already have the `user_id` information retrieved from `req.session`.  This code is unnecessary

File: views/activity.handlebars [hunk 6]
```diff
@@ -39,14 +39,16 @@
                 {{#if (is_attending user_id activity.attending)}}
                   <a class="btn" href="#">Decline</a>
                 {{else}}
-                  <a class="btn" href="#">Attend</a>
+                  <a class="btn" id="attendButton" href="#">Attend</a>
+                  <span class="">{{count}} attending</span>
```
Comment (jdpasternak) [inline views/activity.handlebars:43]: what this key's name is changed in the `homepage-routes`, this `count` should change as well

**Review by jdpasternak — CHANGES_REQUESTED** [review]
Various changes to be made. Please review my comments. Feel free to slack me with questions

File: controllers/api/activity-routes.js (hunk 1 above)
Comment (Naomilounsbury) [reply controllers/api/activity-routes.js:156]: I'll move it to the homeroutes

File: controllers/api/user-routes.js (hunk 2 above)
Comment (Naomilounsbury) [reply controllers/api/user-routes.js:170]: https://stackoverflow.com/questions/54881840/express-session-doesnt-save

File: controllers/homepage-routes.js (hunk 3 above)
Comment (Naomilounsbury) [reply controllers/homepage-routes.js:202]: Ill delete that, it was to make sure that one was rendering not the other

File: controllers/homepage-routes.js (hunk 4 above)
Comment (Naomilounsbury) [reply controllers/homepage-routes.js:207]: fine but this way is also functional

File: views/activity.handlebars (hunk 6 above)
Comment (Naomilounsbury) [reply views/activity.handlebars:43]: fine

File: controllers/api/user-routes.js (hunk 2 above)
Comment (jdpasternak) [reply controllers/api/user-routes.js:170]: This doesn't save the session.  It works without this line

File: controllers/api/activity-routes.js (hunk 1 above)
Comment (jdpasternak) [reply controllers/api/activity-routes.js:156]: If you're writing an API endpoint, it should stay in the API controllers.  What I meant was that this route should strictly response with JSON data

File: controllers/homepage-routes.js (hunk 4 above)
Comment (jdpasternak) [reply controllers/homepage-routes.js:207]: You're right. I'm referring to the consistency in the code.  We want to try to maintain consistency even with components that are not directly related.

File: public/javascript/attend.js (hunk 5 above)
Comment (Naomilounsbury) [reply public/javascript/attend.js:14]: ok deleted

File: controllers/api/activity-routes.js (hunk 1 above)
Comment (Naomilounsbury) [reply controllers/api/activity-routes.js:156]: So it wont refresh with just the json data. re rendering the page refreshes the page and shows the data whenever someone clicks attend

File: controllers/homepage-routes.js (hunk 3 above)
Comment (jdpasternak) [reply controllers/homepage-routes.js:211]: This hasn't been removed

File: controllers/api/user-routes.js (hunk 2 above)
Comment (jdpasternak) [reply controllers/api/user-routes.js:170]: I just tested this and it works without needing `return`. Please remove this before the PR is merged

File: controllers/api/activity-routes.js (hunk 1 above)
Comment (jdpasternak) [reply controllers/api/activity-routes.js:156]: API routes do not render HTML. They should only response with JSON, XML, or some other structured data.  The refreshing of the page should be done on the front end.  After a `fetch` request to `/api/activities/attend`, use `location.reload()` to refresh the page

**Review by jdpasternak — CHANGES_REQUESTED** [review]

**Review by jdpasternak — APPROVED** [review]

**Pull request merged**